test/Libraries/DynamoMSOfficeTests/ExcelTests.cs

```diff
@@ -983,6 +979,247 @@ public void TestWriteFunctionObjectToExcel()
+            //3.  Rerun the file 
+            
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteModifyGraph_2.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            ViewModel.HomeSpace.Run();
+            var watch = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.ReadFromFile");
+            var watch2 = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.WriteToFile");
+            var watch3 = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("17bf44dd-0285-496f-a388-58649cadbff8");
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            //from excel - {5,6,7,8,9,10}
+            var data = new object[] { new object[] {5}, new object[] { 6 }, new object[] { 7 }, new object[] { 8 }, new object[] { 9 }, new object[] { 10 } };
+            AssertPreviewValue(watch.GUID.ToString(), data);
+
+            // decrease the data written into excel
+            // Write {2,3,4,5} into excel 
+            ConnectorModel.Make(watch3, watch2, 0, 4);
+            
+            ViewModel.HomeSpace.Run();
+
+            Assert.IsTrue(watch2.CachedValue.IsCollection);
+            
+            var data2 = new object[] { new object[] { 2}, new object[] { 3 }, new object[] { 4 }, new object[] { 5 }, new object[] { null}, new object[] { null} };
+            AssertPreviewValue(watch2.GUID.ToString(), data2);
+
+        }
+
+        [Test]
+        public void WriteModifyGraph_3()
+        {
+            //incomplete
+            // 1.  Write into  an Existing excel file and Run
+            // 2.  Increase the array size written to an excel file
+            // 3.  Rerun the file 
+
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteModifyGraph_3.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            ViewModel.HomeSpace.Run();
+            var watch = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.ReadFromFile");
+            var watch2 = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.WriteToFile");
+            // Write {5,6,7,8,9,10} into excel
+            var watch3 = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("17bf44dd-0285-496f-a388-58649cadbff8");
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            var list1 = watch.CachedValue.GetElements();
```

Comment:
ive resolved this in commit -https://github.com/monikaprabhu/Dynamo/commit/554878155adb111820d98d40b2f0e04325d6431e